cinema/views.py

```diff
@@ -19,26 +22,39 @@
     OrderSerializer,
     OrderListSerializer,
 )
+from cinema_service.permissions import IsAdminOrIfAuthenticatedReadOnly
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(mixins.ListModelMixin, mixins.CreateModelMixin,
+                   GenericViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
to avoid code duplication add TokenAuthentication to the default authentication_classes in settings.py and add IsAdminOrIfAuthenticatedReadOnly to the default permission_classes in settings.py